src/objects/Mesh.js

```diff
@@ -210,7 +212,12 @@ Mesh.prototype = Object.assign( Object.create( Object3D.prototype ), {
 
 				}
 
-				intersection.face = new Face3( a, b, c, Triangle.normal( vA, vB, vC ) );
+				var face = new Face3( a, b, c );
+
+				triangle.set( vA, vB, vC ).normal( faceNormal );
```

Comment:
@WestLangley In context of `BufferGeometry.center()`, the semantics of `Triangle.normal()` is also problematic. It should be `Triangle.getNormal()`...

Reply:
Yes, I guess it should be `Triangle.getNormal( a, b, c, target )` and `Triangle.prototype.getNormal( a, b, c, target )`.

Also `barycoordFromPoint()` -> `getBarycoord()`.

Separate PR, though.